src/legacy/ui/public/vis/editors/default/components/default_editor_agg.test.tsx

```diff
@@ -0,0 +1,282 @@
+      const actions = shallow(comp.prop('extraAction'));
+
+      expect(actions.children().length).toBe(2);
+      expect(actions.childAt(0).props()).toHaveProperty('content', 'Disable aggregation');
+      expect(actions.childAt(1).props()).toHaveProperty('content', 'Remove dimension');
+    });
+
+    it('should have draggable action', () => {
+      defaultProps.isDraggable = true;
+      const comp = shallow(<DefaultEditorAgg {...defaultProps} />);
+      const actions = shallow(comp.prop('extraAction'));
+
+      expect(actions.children().length).toBe(1);
+      expect(actions.childAt(0).props()).toHaveProperty('content', 'Modify priority by dragging');
+      expect(actions.childAt(0).props()).toHaveProperty('type', 'grab');
+    });
+
+    it('should disable agg', () => {
+      defaultProps.isRemovable = true;
+      const comp = mount(<DefaultEditorAgg {...defaultProps} />);
+      comp.find('[iconType="eye"]').simulate('click');
+
+      expect(defaultProps.onToggleEnableAgg).toBeCalledWith(defaultProps.agg, false);
+    });
+
+    it('should enable agg', () => {
+      defaultProps.agg.enabled = false;
+      const comp = mount(<DefaultEditorAgg {...defaultProps} />);
+      comp.find('[iconType="eyeClosed"]').simulate('click');
+
+      expect(defaultProps.onToggleEnableAgg).toBeCalledWith(defaultProps.agg, true);
+    });
+
+    it('should call removeAgg', () => {
+      defaultProps.isRemovable = true;
+      const comp = mount(<DefaultEditorAgg {...defaultProps} />);
+      comp.find('[iconType="cross"]').simulate('click');
+
+      expect(defaultProps.removeAgg).toBeCalledWith(defaultProps.agg);
+    });
+  });
+
+  describe('last bucket', () => {
+    beforeEach(() => {
+      defaultProps.isLastBucket = true;
+      defaultProps.lastParentPipelineAggTitle = 'ParentPipelineAgg';
+    });
+
+    it('should disable min_doc_count when agg is histogram or date_histogram', () => {
+      defaultProps.agg.type = {
+        name: 'histogram',
+      };
+      const compHistogram = shallow(<DefaultEditorAgg {...defaultProps} />);
+      defaultProps.agg.type = {
+        name: 'date_histogram',
+      };
+      const compDateHistogram = shallow(<DefaultEditorAgg {...defaultProps} />);
+
+      expect(compHistogram.find('DefaultEditorAggParams').props()).toHaveProperty(
+        'disabledParams',
+        ['min_doc_count']
+      );
+      expect(compDateHistogram.find('DefaultEditorAggParams').props()).toHaveProperty(
+        'disabledParams',
+        ['min_doc_count']
+      );
+    });
+
+    it('should set error when agg is not histogram or date_histogram', () => {
+      defaultProps.agg.type = {
+        name: 'aggType',
+      };
+      const comp = shallow(<DefaultEditorAgg {...defaultProps} />);
+
+      expect(comp.find('DefaultEditorAggParams').props()).toHaveProperty(
+        'aggError',
+        'Last bucket aggregation must be "Date Histogram" or "Histogram" when using "ParentPipelineAgg" metric aggregation.'
```

Comment:
I suppose it's not correct to check the whole error text string since it could be changed in the future.

Reply:
Updated.